SDL2 fails to build on ARM64 on Windows

#### System Information
- Ogre Version: v14.4.1
- Operating System / Platform: Windows 11 / ARM64
- RenderSystem: NA

#### Detailled description
When configuring ogre on ARM64 on Windows I'm hitting linking errors about unresolved `_Interlocked*` symbols in SDL2. E.g.:

`SDL_malloc.c.obj : error LNK2001: unresolved external symbol _InterlockedCompareExchange`

They are caused by an [issue in SDL](https://github.com/libsdl-org/SDL/issues/13254) that has been fixed in the most recent version. I'll put up a PR soon to upgrade SDL.

--- a/CMake/Dependencies.cmake
+++ b/CMake/Dependencies.cmake
@@ -129,16 +129,16 @@ if(OGRE_BUILD_DEPENDENCIES AND NOT EXISTS ${OGREDEPS_PATH})
     if(MSVC OR MINGW OR SKBUILD) # other platforms dont need this
         message(STATUS "Building SDL2")
         file(DOWNLOAD
-            https://libsdl.org/release/SDL2-2.32.8.tar.gz
-            ${PROJECT_BINARY_DIR}/SDL2-2.32.8.tar.gz)
+            https://libsdl.org/release/SDL2-2.32.10.tar.gz
+            ${PROJECT_BINARY_DIR}/SDL2-2.32.10.tar.gz)
         execute_process(COMMAND ${CMAKE_COMMAND} 
-            -E tar xf SDL2-2.32.8.tar.gz WORKING_DIRECTORY ${PROJECT_BINARY_DIR})
+            -E tar xf SDL2-2.32.10.tar.gz WORKING_DIRECTORY ${PROJECT_BINARY_DIR})
         execute_process(COMMAND ${CMAKE_COMMAND}
             -E make_directory ${PROJECT_BINARY_DIR}/SDL2-build)
         execute_process(COMMAND ${BUILD_COMMAND_COMMON}
             -DSDL_STATIC=FALSE
             -DCMAKE_INSTALL_LIBDIR=lib
-            ${PROJECT_BINARY_DIR}/SDL2-2.32.8
+            ${PROJECT_BINARY_DIR}/SDL2-2.32.10
             WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/SDL2-build)
         execute_process(COMMAND ${CMAKE_COMMAND}
             --build ${PROJECT_BINARY_DIR}/SDL2-build ${BUILD_COMMAND_OPTS})

